Problem running cocotb with Questa and Modelsim

I am trying to run a Lattice IP simulation using Questa or Modelsim. This problem that I will explain will be for Questa, but the same problem is present in Modelsim.
Here is the terminal output after running just a simple cocotb testbench for a Multiplier IP.
```
make test SIMULATOR=questa TOPLEVEL=top MODULE=multiplier_tb
make[1]: Entering directory '/home/user/SDR-HLS/1.RTLImplementation/2.sim/Multiplier'
make -C . SIM=questa TOPLEVEL_LANG=verilog TOPLEVEL=top MODULE=multiplier_tb
make[2]: Entering directory '/home/user/SDR-HLS/1.RTLImplementation/2.sim/Multiplier'
rm -f results.xml
make -f Makefile results.xml
make[3]: Entering directory '/home/user/SDR-HLS/1.RTLImplementation/2.sim/Multiplier'
mkdir -p sim_build
# Make sure all libs in SOURCES_VHDL_* are mentioned in VHDL_LIB_ORDER and vice versa
rm -f results.xml
set -o pipefail; MODULE=multiplier_tb TESTCASE= TOPLEVEL=top \
GPI_EXTRA= TOPLEVEL_LANG=verilog \
 /home/user/FPGA/tools/radiant/questasim/bin/vsim -c -64  -do sim_build/runsim.do  2>&1 | tee sim_build/sim.log
Reading pref.tcl

# 2024.2

# do sim_build/runsim.do
# Questa Lattice OEM Edition-64 vmap 2024.2 Lib Mapping Utility 2024.05 May 20 2024
# vmap -c 
# Copying /home/user/FPGA/tools/radiant/questasim/linux_x86_64/../modelsim.ini to modelsim.ini
# Questa Lattice OEM Edition-64 vmap 2024.2 Lib Mapping Utility 2024.05 May 20 2024
# vmap work sim_build/work 
# Modifying modelsim.ini
# vsim -onfinish exit -pli "/home/user/.local/lib/python3.10/site-packages/cocotb/libs/libcocotbvpi_modelsim.so" sim_build/top 
# Start time: 15:24:36 on Sep 11,2024
# ** Note: (vsim-3812) Design is being optimized...
# ** Warning: (vsim-3865) Due to PLI being present, global +acc is being enabled automatically. This will cause your simulation to run very slowly. Please use vsim -no_autoacc to disable this feature. This feature is now deprecated and will be removed from future releases.
# ** Warning: (vopt-10587) Some optimizations are turned off because the +acc switch is in effect. This will cause your simulation to run slowly. Please use -access/-debug to maintain needed visibility. The +acc switch would be deprecated in a future release.
# ** Error: (vopt-11) Could not find work.sim_build/top.
# Optimization failed
# ** Note: (vsim-12126) Error and warning message counts have been restored: Errors=1, Warnings=2.
# Error loading design
Error loading design
# End time: 15:24:37 on Sep 11,2024, Elapsed time: 0:00:01
# Errors: 1, Warnings: 2
make[3]: *** [/home/user/.local/lib/python3.10/site-packages/cocotb/share/makefiles/simulators/Makefile.questa:160: results.xml] Error 12
make[3]: Leaving directory '/home/user/SDR-HLS/1.RTLImplementation/2.sim/Multiplier'
make[2]: *** [/home/user/.local/lib/python3.10/site-packages/cocotb/share/makefiles/Makefile.inc:40: sim] Error 2
make[2]: Leaving directory '/home/user/SDR-HLS/1.RTLImplementation/2.sim/Multiplier'
make[1]: *** [Makefile:39: test] Error 2
make[1]: Leaving directory '/home/user/SDR-HLS/1.RTLImplementation/2.sim/Multiplier'
make: *** [Makefile:44: run] Error 2

```
The main problem is detailed in this line
```
# ** Error: (vopt-11) Could not find work.sim_build/top.
```

Versions:
cocotb: Version: 1.8.1
OS: Ubuntu 22.04 LTS x86_64
Simulator: Questa Lattice OEM Edition-64 vsim 2024.2 Simulator 2024.05 May 20 2024
Python: Python 3.10.12

--- a/src/cocotb_tools/makefiles/simulators/Makefile.questa
+++ b/src/cocotb_tools/makefiles/simulators/Makefile.questa
@@ -139,7 +139,7 @@ endif
 ifdef SCRIPT_FILE
 	@echo "do $(SCRIPT_FILE)" >> $@
 endif
-	@echo "vsim $(VSIM_ARGS) $(EXTRA_ARGS) $(call deprecate,PLUSARGS,COCOTB_PLUSARGS) $(SIM_BUILD)/$(COCOTB_TOPLEVEL)" >> $@
+	@echo "vsim $(VSIM_ARGS) $(EXTRA_ARGS) $(call deprecate,PLUSARGS,COCOTB_PLUSARGS) $(COCOTB_TOPLEVEL)" >> $@
 ifeq ($(WAVES),1)
 	@echo "log -recursive /*" >> $@
 endif

--- a/tests/test_cases/test_iteration_vhdl/test_iteration.py
+++ b/tests/test_cases/test_iteration_vhdl/test_iteration.py
@@ -27,37 +27,17 @@ import logging
 import os
 
 import cocotb
-from cocotb._sim_versions import QuestaVersion
 from cocotb.triggers import Combine, Timer
 
 
 def total_object_count():
     """Return the total object count based on simulator."""
     SIM_NAME = cocotb.SIM_NAME.lower()
-    SIM_VERSION = cocotb.SIM_VERSION.lower()
 
     # Questa with VHPI
     # TODO: Why do we get massively different numbers for Questa/VHPI than for Questa/FLI or VPI?
     if SIM_NAME.startswith("modelsim") and os.environ["VHDL_GPI_INTERFACE"] == "vhpi":
         return 5119
-
-    if SIM_NAME.startswith("modelsim") and os.environ["VHDL_GPI_INTERFACE"] == "fli":
-        # Questa 2024.1 onwards (FLI) do not discover the following objects, which
-        # are instantiated four times:
-        # - inst_generic_sp_ram.clk (<class 'cocotb.handle.LogicObject'>)
-        # - inst_generic_sp_ram.rst (<class 'cocotb.handle.LogicObject'>)
-        # - inst_generic_sp_ram.wen (<class 'cocotb.handle.LogicObject'>)
-        if QuestaVersion(SIM_VERSION) >= QuestaVersion("2024.1"):
-            return 2663 - 4 * 3
-
-        # Questa 2023.1 onwards (FLI) do not discover the following objects, which
-        # are instantiated four times:
-        # - inst_generic_sp_ram.clk (<class 'cocotb.handle.LogicObject'>)
-        # - inst_generic_sp_ram.rst (<class 'cocotb.handle.LogicObject'>)
-        # - inst_generic_sp_ram.wen (<class 'cocotb.handle.LogicObject'>)
-        # - inst_generic_sp_ram.en (<class 'cocotb.handle.LogicObject'>)
-        if QuestaVersion(SIM_VERSION) >= QuestaVersion("2023.1"):
-            return 2663 - 4 * 4
 
     if SIM_NAME.startswith(
         (

